doc/source/whatsnew/v0.23.0.txt

```diff
@@ -876,3 +876,4 @@ Other
 ^^^^^
 
 - Improved error message when attempting to use a Python keyword as an identifier in a ``numexpr`` backed query (:issue:`18221`)
+- Bug in :func:`pandas.core.DictWrapper.__getattr__` when looking up a non-existent key (:issue:`19789`)
```

Comment:
Could you describe the bug a bit? Mention it raising a `KeyError` instead of the correct `OptionError`.